src/main/java/org/opensearch/knn/index/codec/BasePerFieldKnnVectorsFormat.java

```diff
@@ -30,6 +41,7 @@ public abstract class BasePerFieldKnnVectorsFormat extends PerFieldKnnVectorsFor
     private final int defaultBeamWidth;
     private final Supplier<KnnVectorsFormat> defaultFormatSupplier;
     private final BiFunction<Integer, Integer, KnnVectorsFormat> formatSupplier;
+    private final Function5Arity<Integer, Integer, Float, Integer, Boolean, KnnVectorsFormat> quantizedVectorsFormatSupplier;
```

Comment:
this is complete abuse of Functional interface, it provides no understanding what is the meaning of `Integer, Integer, Float, Integer, Boolean`. Put it under a class and use that class.